source/main/index.js

```diff
@@ -96,6 +98,30 @@ const onAppReady = async () => {
   const isInSafeMode = includes(process.argv.slice(1), '--safe-mode');
 
   mainWindow = createMainWindow(isInSafeMode);
+
+  const onCheckDiskSpace = ({ notEnoughSpace }: CheckDiskSpaceResponse) => {
+    if (cardanoNode) {
+      if (notEnoughSpace) {
+        try {
+          cardanoNode.stop();
```

Comment:
@daniloprates @DominikGuzei we should perhaps check if cardanoNode is already stopped before trying to stop it again. That way we would avoid many stopping attempts: 

![screen shot 2019-01-08 at 13 12 06](https://user-images.githubusercontent.com/376611/50830161-41db2f00-1347-11e9-9e94-bc547fec5090.png)

Reply:
This probably due to our automatic restart feature which is used during initial Daedalus start.
In case cardano-node exits Daedalus will try to start it again for 5 consecutive times before giving up. Not sure how we can bypass that in this case. Help appreciated @DominikGuzei 🙏

Reply:
@daniloprates I guess your original code simply shuts it down every of these 5 restart attempts. Which makes it look like it is working BUT we should avoid even trying to restart it if we have shut it down due to not-enough disk space.

Reply:
Not sure how to deal with it. I have added the [condition](https://github.com/input-output-hk/daedalus/pull/1157/commits/b1b24475d2c54294d464e00f3722b0c5aac30155#diff-286decdfc7881f7334e73465dab511ffR106), but as the node restarts (it won't trigger the `node.restartNode` nor the `node.start` functions, though), it's still calling `node.stop` every 10 secs.

Reply:
I will sort this out @daniloprates - no worries!

Reply:
@daniloprates fixed in: https://github.com/input-output-hk/daedalus/pull/1157/commits/245017e19ea98e06e99b61460cdc5768228b9ec3